# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.cpp
# Commit: d1558d374cef
# Full Hash: d1558d374cef9d954a84b7db4646e48a4878f621
# Author: James Teh <jteh@mozilla.com>
# Date: 2023-11-30 04:12:04
# Regressor Bug: 1862802
# File Overlap Count: 1
# Description:
#   Bug 1862802 part 1: Provide an argument to PlatformCaretMoveEvent specifying whether the event was caused by user input. r=eeejay
#   
#   This is needed in order to support find in page on Android using remote caret events instead of virtual cursor change events.
#   
#   Depends on D192641
# ==============================================================================

diff -r d6f61c448b90 -r d1558d374cef accessible/generic/LocalAccessible.cpp
--- a/accessible/generic/LocalAccessible.cpp	Tue Nov 28 21:50:11 2023 +0000
+++ b/accessible/generic/LocalAccessible.cpp	Tue Nov 28 22:03:19 2023 +0000
@@ -915,7 +915,8 @@
           AccCaretMoveEvent* event = downcast_accEvent(aEvent);
           ipcDoc->SendCaretMoveEvent(
               id, event->GetCaretOffset(), event->IsSelectionCollapsed(),
-              event->IsAtEndOfLine(), event->GetGranularity());
+              event->IsAtEndOfLine(), event->GetGranularity(),
+              event->IsFromUserInput());
           break;
         }
         case nsIAccessibleEvent::EVENT_TEXT_INSERTED:
@@ -1038,9 +1039,9 @@
       // AccessibleWrap::UpdateSystemCaretFor currently needs to call
       // HyperTextAccessible::GetCaretRect again to get the widget and there's
       // no point calling it twice.
-      PlatformCaretMoveEvent(target, event->GetCaretOffset(),
-                             event->IsSelectionCollapsed(),
-                             event->GetGranularity(), rect);
+      PlatformCaretMoveEvent(
+          target, event->GetCaretOffset(), event->IsSelectionCollapsed(),
+          event->GetGranularity(), rect, event->IsFromUserInput());
       break;
     }
     case nsIAccessibleEvent::EVENT_TEXT_INSERTED: